tests/phpunit/tests/blocks/register.php

```diff
@@ -322,6 +322,73 @@ public function test_wrong_array_index_do_not_register_block_style_handle() {
 		$this->assertFalse( $result );
 	}
 
+	/**
+	 * @ticket 58605
+	 * @dataProvider data_register_block_style_handle_uses_correct_core_stylesheet
+	 *
+	 * @param string      $block_json_path Path to the `block.json` file, relative to ABSPATH.
+	 * @param string      $style_field     Either 'style' or 'editorStyle'.
+	 * @param string|bool $expected_path   Expected path of registered stylesheet, relative to ABSPATH.
+	 */
+	public function test_register_block_style_handle_uses_correct_core_stylesheet( $block_json_path, $style_field, $expected_path ) {
+		$metadata_file = ABSPATH . $block_json_path;
+		$metadata      = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
+
+		$block_name = str_replace( 'core/', '', $metadata['name'] );
+
+		// Normalize metadata similar to `register_block_type_from_metadata()`.
+		$metadata['file'] = wp_normalize_path( realpath( $metadata_file ) );
+		if ( ! isset( $metadata['style'] ) ) {
+			$metadata['style'] = "wp-block-$block_name";
```

Comment:
```suggestion
			$metadata['style'] = "wp-block-{$block_name}";
```

Reply:
This is the same logic `register_block_type_from_metadata`. I don't think we should change this.

Reply:
Doesn't make any functional difference, I don't feel strongly either way. Maybe better to just keep as is to match `register_block_type_from_metadata()`.